kobo/apps/superuser_stats/views.py

```diff
@@ -22,40 +31,227 @@ def _base_filename_to_full_filename(base_filename, username):
 
 @user_passes_test(lambda u: u.is_superuser)
 def country_report(request):
+    """
+    Generates a report which counts the number of submissions and forms
+    per country as reported by the user
+    """
+    today = timezone.now().date()
     base_filename = 'country-report_{}_{}_{}.csv'.format(
         re.sub('[^a-zA-Z0-9]', '-', request.META['HTTP_HOST']),
-        date.today(),
+        today,
         datetime.now().microsecond
     )
 
-    # Get the date date filters from the query and set defaults
-    start_date = request.GET.get('start_date', date.today())
-    tomorrow = date.today() + timedelta(days=1)
+    # Get the date filters from the query and set defaults
+    start_date = request.GET.get('start_date', today)
+    tomorrow = today + timedelta(days=1)
     end_date = request.GET.get('end_date', tomorrow)
 
     # Generate the CSV file
     filename = _base_filename_to_full_filename(
         base_filename, request.user.username)
     generate_country_report.delay(filename, start_date, end_date)
 
+    url = f"{KOBOFORM_URL}{reverse('superuser_stats:countries_report')}"
+    template_ish = (
+        f'<html><head><title>Countries report</title></head>'
+        f'<body>Your report is being generated. Once finished, it will be '
+        f'available at <a href="{base_filename}">{base_filename}</a>.<br>'
+        f'If you receive a 404, please refresh your browser periodically until '
+        f'your request succeeds.<br><br>'
+        f'To select a date range, add a <code style="background: lightgray">?</code> at the end of the URL and set the '
+        f'<code style="background: lightgray">start_date</code> parameter to <code style="background: lightgray">YYYY-MM-DD</code> and/or the '
+        f'<code style="background: lightgray">end_date</code> parameter to <code style="background: lightgray">YYYY-MM-DD</code>.<br><br>'
+        f'<b>Example:</b><br>'
+        f'<a href="{url}?start_date=2020-01-31&end_date=2021-02-28">'
+        f'  {url}?start_date=2020-01-31&end_date=2021-02-28'
+        f'</a>'
+        f'</body></html>'
+    )
+
+    return HttpResponse(template_ish)
+
+
+@user_passes_test(lambda u: u.is_superuser)
+def continued_usage_report(request):
+    """
+    Tracks users usage over a given year with the last day being set by
+    'end_date'
+    """
+    today = timezone.now().date()
+    base_filename = 'continued-usage-report_{}_{}_{}.csv'.format(
+        re.sub('[^a-zA-Z0-9]', '-', request.META['HTTP_HOST']),
+        today,
+        timezone.now().microsecond
+    )
+
+    # Get the date filters from the query and set defaults
+    tomorrow = today + timedelta(days=1)
+    end_date = request.GET.get('end_date', tomorrow)
+
+    # Generate the CSV file
+    filename = _base_filename_to_full_filename(
+        base_filename, request.user.username)
+    generate_continued_usage_report.delay(filename, end_date)
```

Comment:
@joshuaberetta, nice catch for the `Date joined` column.
The submissions are there. The counters were not populated (KC needs to be on branch `daily-and-monthly-xform-counters-management-command`). I'll update the PR description

Reply:
Great — I see the counts showing up now 🎉 maybe it would also be good to show a count of `0` for submissions like we're doing for assets to be consistent? (pretty minor though)
So from this:

Username | Date Joined | Last Login | Assets 3m | Assets 6m | Assets 12m | Submissions 3m | Submissions 6m | Submissions 12M
-- | -- | -- | -- | -- | -- | -- | -- | --
joshuaberetta | 2021-07-07 18:38:10.161650+00:00 | 2022-10-06 13:38:43.515195+00:00 | 0 | 0 | 3 |   |   | 6

To this:

Username | Date Joined | Last Login | Assets 3m | Assets 6m | Assets 12m | Submissions 3m | Submissions 6m | Submissions 12M
-- | -- | -- | -- | -- | -- | -- | -- | --
joshuaberetta | 2021-07-07 18:38:10.161650+00:00 | 2022-10-06 13:38:43.515195+00:00 | 0 | 0 | 3 | 0  |   0 | 6

Reply:
Additionally, if I try the suggested example of date filtering (https://kf.tri.kbtdev.org/superuser_stats/reports/continued-usage/?end_date=2021-02-28) I get the `Not Found The requested resource was not found on this server.` again even if I wait a while and refresh — I know you addressed this [here](https://github.com/kobotoolbox/kpi/pull/3770#discussion_r989114373), but just pointing it out again 🙂

Reply:
(empty cells) Fixed as suggested! 
No reports are generated if no data match date range.